jekyll/_posts/2018-06-09-discover-godot-nim.md

```diff
@@ -0,0 +1,93 @@
+Programming in Nim is fun for me, because of the clear syntax, fast compile times and helpful community. I haven't even gotten to the juicy parts of Nim: metaprogramming!
+
+```nim
+# This is my solution for the first Rosalind problem
+# also one of the first Nim snippets I wrote
+import tables
+
+const input = "AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGAT"
+var symbolCount = initCountTable[char]()
+
+for c in input:
+    symbolCount.inc(c, 1)
+
+for k, v in symbolCount:
+    stdout.write v, " "
+```
+
+#####
+
+### Experimenting with Godot
+
+Now I don't have that much experience with Godot as I've recently started using it. The [current documentation](http://docs.godotengine.org/en/3.0/) the engine provides is great and it covers everything you need to get started making a game.
+At first I found Godot quite intimidating(so many bells and whistles!), but browsing through the docs(and google) solved most of my questions. Godot also has an incredible helpful and awesome community, reachable through [multiple channels](https://godotengine.org/community). Open source at it's best.
+Thinking in nodes is the first thing I had to learn. Everything in Godot is a node and a tree of nodes make up a scene. Your game could be one scene or multiple scenes, it depends on how you tackle the design of your game. Personally I find it a lot easier to make a project in Godot if I make a rough sketch on paper of all the important nodes in a game and how they work together.
+
+### Godot + Nim = ???
+
+Somewhere halfway I stopped using GDScript and started my journey on figuring out how the hell the [Nim bindings](https://github.com/pragmagic/godot-nim) work. This... wasn't the smartest of choices as there aren't many examples on how to work with the bindings. The [godotnim-stub](https://github.com/pragmagic/godot-nim-stub) is an excellent starting place if you have previous experience, but as a beginner I was continously searching for hints, even looking through other GDNative bindings trying to find a way to do something which is so simple in GDScript. Looking back I definitely would be more productive in GDScript and I'd understand Godot alot better.
+_But!_ I did finish 2 small samples following the ["Your first game"](http://docs.godotengine.org/en/3.0/getting_started/step_by_step/your_first_game.html) tutorial and a [Conways Game of Life](http://www.brainjargames.com/game-of-life/) post using Nim instead of GDScript. Using types made things a lot clearer in code and I learned the awesome feeling of tackling a problem that might seem impossible to figure out at first, but to which the answer is so simple once solved.
+For those who are also keen to try Nim with Godot, I'd say take a look at the stub project and the api the bindings generate. My experience following the GDScript focused tutorials with Nim was that it all translates pretty well. Nim compiles so fast that I didn't find myself growing tired of the compile script -> press f5 in godot cycle.
```

Comment:
~~api~~ → API
~~growing tired of the compile script -> press f5 in godot cycle~~ → growing tired of the "compile script -> press F5 in Godot" cycle